### ISSUE TITLE
why do you use Keycloak

### ISSUE THREAD

**Issue by HelloTo4x** [ivangfr/springboot-keycloak-openldap#6]
Hi, can u tell me why u use Keycloak instead of Spring security, which feature does it have but Spring security doesn't have it ?

**Comment by HelloTo4x** [comment]
i have a question, why Spring security isn't an authorization ? https://docs.spring.io/spring-security/reference/servlet/authorization/index.html , i see it can authorizate by @PreAuthor, ... or something else like this

**Comment by ivangfr** [comment]
Ok, I got your point. Sure, we can use just Spring Security but then, we will need to handle the A&A of the application users in the app itself, maybe saving the users' username, password and roles in a database, for instance. On the other hand, when using Keycloak, all the users' management can be handled by Keycloak. In our app, we just need to focus on the logic the app must deliver. So, using Keycloak, we externalize the users' management. It can benefit not only this app, but other that need security also.

**Comment by HelloTo4x** [comment]
thank you so much

**Comment by ivangfr** [comment]
Not exactly. When we are handling security, we need to work with **authentication** (verify who someone is) and **authorization** (verify what someone can access).

**Comment by ivangfr** [comment]
@HelloTo4x btw, have a look at these projects
- https://github.com/ivangfr/springboot-react-basic-auth
- https://github.com/ivangfr/springboot-react-jwt-token
- https://github.com/ivangfr/springboot-react-social-login

On them, I've used just Spring Security. For instance, in `springboot-react-basic-auth`, I've created a table to save the users of the application; on `springboot-react-social-login`, you can access the app using `Github` account.
